### ISSUE TITLE
Segmentation Fault (core dumped) when running build_model(configs.ner.ner_rus, download=True)

### ISSUE THREAD

**Issue by hpylieva** [deepmipt/DeepPavlov#824]
Hi! I get `Segmentation Fault (core dumped)` when running `build_model(configs.ner.ner_rus, download=True)`. The version of deeppavlov I use is 0.2.0

**Comment by yoptar** [comment]
Hi!
Were there any other output messages before the `Segmentation Fault`? Also, could you provide info on the system that was used?

**Comment by hpylieva** [comment]
Hi @yoptar , no other messages, just this one and an immediate exit from the python3 process. System: Ubuntu 18.04.2

**Comment by yoptar** [comment]
Are you sure that there was nothing? Not even logs about downloading models files?
Can you also say what version of python you're using, what packages are installed (the result of the `pip list` command for example) and some info on your hardware: CPU and GPU if any.

**Comment by hpylieva** [comment]
Solved the problem by installing the package in the virtual environment as recommended and running `build_model` twice. Twice - because at first, the process ended with `ModuleNotFoundError: No module named 'fastText'` error. Here is the full log of the first run:
```
>>> build_model(configs.ner.ner_rus, download=True)
2019-04-30 21:18:49.374 INFO in 'deeppavlov.download'['download'] at line 116: Skipped http://files.deeppavlov.ai/embeddings/lenta_lower_100.bin download because of matching hashes
2019-04-30 21:18:49.563 INFO in 'deeppavlov.download'['download'] at line 116: Skipped http://files.deeppavlov.ai/deeppavlov_data/ner_rus_v2_cpu_compatible.tar.gz download because of matching hashes
[nltk_data] Downloading package punkt to /home/h.pylieva/nltk_data...
[nltk_data]   Package punkt is already up-to-date!
[nltk_data] Downloading package stopwords to
[nltk_data]     /home/h.pylieva/nltk_data...
[nltk_data]   Package stopwords is already up-to-date!
[nltk_data] Downloading package perluniprops to
[nltk_data]     /home/h.pylieva/nltk_data...
[nltk_data]   Package perluniprops is already up-to-date!
[nltk_data] Downloading package nonbreaking_prefixes to
[nltk_data]     /home/h.pylieva/nltk_data...
[nltk_data]   Package nonbreaking_prefixes is already up-to-date!
2019-04-30 21:18:50.320 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/word.dict]
2019-04-30 21:18:50.359 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/tag.dict]
2019-04-30 21:18:50.360 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/char.dict]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/commands/infer.py", line 61, in build_model
    component = from_params(component_config, mode=mode, serialized=component_serialized)
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 94, in from_params
    cls = get_model(cls_name)
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/common/registry.py", line 71, in get_model
    return cls_from_str(_REGISTRY[name])
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/common/registry.py", line 40, in cls_from_str
    return getattr(importlib.import_module(module_name), cls_name)
  File "/home/h.pylieva/env/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 994, in _gcd_import
  File "<frozen importlib._bootstrap>", line 971, in _find_and_load
  File "<frozen importlib._bootstrap>", line 955, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 665, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/models/embedders/fasttext_embedder.py", line 18, in <module>
    import fastText
ModuleNotFoundError: No module named 'fastText'
```
Then I just rerun the command but in slightly changed form:
```
>> model = build_model(configs.ner.ner_rus, download=True)
```
and for this time the process ended with `ModuleNotFoundError: No module named 'gensim'`. I installed gensim, then tensorflow (same, after an ModuleNotFoundError) and the model finally was loaded.

### What I did differently previously
When I ran `build_model` previously (locally, not in venv) the situation with requiring 'fastText' was exactly the same. So I installed `fastText`. Then I noticed that the current version of 'fasttext' seems to be with all letters in lowercase:
```
$ pip3 freeze | grep 'fasttext'
fasttext==0.8.3
```
whereas in deeppavlov files we have ```import fastText``` and for this reason, my installed package wasn't recognized. That is why I changed this line in line to the the following:
```import fasttext as fastText```
and then after rerun of `build_model` received the `Segmentation Fault`.

**Comment by hpylieva** [comment]
Hi again. I found out that successful run for me was reached for a slightly another command```build_model(configs.ner.ner_ontonotes)```. With ```build_model(configs.ner.ner_rus)``` I still have the same problems: unable to find fastText when ``` [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/char.dict]``` and after in ```deeppavlov/models/embedders/fasttext_embedder.py``` `import fastText` is substituted with `import fasttext` then I get memory errors:
```
>>> ner_model = build_model(configs.ner.ner_rus)
2019-05-02 12:44:47.99 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/word.dict]
2019-05-02 12:44:47.133 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/tag.dict]
2019-05-02 12:44:47.133 INFO in 'deeppavlov.core.data.simple_vocab'['simple_vocab'] at line 103: [loading vocabulary from /home/h.pylieva/.deeppavlov/models/ner_rus/char.dict]
2019-05-02 12:44:47.134 INFO in 'deeppavlov.models.embedders.fasttext_embedder'['fasttext_embedder'] at line 52: [loading fastText embeddings from `/home/h.pylieva/.deeppavlov/downloads/embeddings/lenta_lower_100.bin`]
2019-05-02 12:44:47.189 ERROR in 'deeppavlov.core.common.params'['params'] at line 110: Exception in <class 'deeppavlov.models.embedders.fasttext_embedder.FasttextEmbedder'>
Traceback (most recent call last):
  File "fasttext/fasttext.pyx", line 152, in fasttext.fasttext.load_model
RuntimeError: vector::_M_default_append

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 104, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/models/embedders/abstract_embedder.py", line 56, in __init__
    self.load()
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/models/embedders/fasttext_embedder.py", line 53, in load
    self.model = fastText.load_model(str(self.load_path))
  File "fasttext/fasttext.pyx", line 154, in fasttext.fasttext.load_model
Exception: fastText: Cannot load /home/h.pylieva/.deeppavlov/downloads/embeddings/lenta_lower_100.bin due to C++ extension failed to allocate the memory
Traceback (most recent call last):
  File "fasttext/fasttext.pyx", line 152, in fasttext.fasttext.load_model
RuntimeError: vector::_M_default_append

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/commands/infer.py", line 61, in build_model
    component = from_params(component_config, mode=mode, serialized=component_serialized)
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/core/common/params.py", line 104, in from_params
    component = cls(**dict(config_params, **kwargs))
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/models/embedders/abstract_embedder.py", line 56, in __init__
    self.load()
  File "/home/h.pylieva/env/lib/python3.6/site-packages/deeppavlov/models/embedders/fasttext_embedder.py", line 53, in load
    self.model = fastText.load_model(str(self.load_path))
  File "fasttext/fasttext.pyx", line 154, in fasttext.fasttext.load_model
Exception: fastText: Cannot load /home/h.pylieva/.deeppavlov/downloads/embeddings/lenta_lower_100.bin due to C++ extension failed to allocate the memory
```
I run this in a separate virtual environment and have 18Gb of available memory:
```
                       total        used        free      shared  buff/cache   available
Mem:            31G         11G        6.1G        1.0G         13G         18G
Swap:          8.0G        264K        8.0G
```

**Comment by sadov-m** [comment]
Hi @yoptar ,
I'm facing the same issue that was described by @hpylieva : 

> ModuleNotFoundError: No module named 'fastText'

I'm using _CentOS Linux release 7.5.1804_ and the following libraries:

- deeppavlov 0.1.6
- gensim 3.7.1
- tensorflow 1.12.0
- fasttext 0.8.3

Could you tell us, please, what we're doing wrong?

**Comment by sadov-m** [comment]
Okay, I successfully ran `build_model(configs.ner.ner_rus)` after the following steps:

- I looked through the _fasttext.txt_ file in `/root/anaconda3/lib/python3.6/site-packages/deeppavlov/requirements` and found the string `git+https://github.com/deepmipt/fastText.git#egg=fastText==0.8.22`; so I tried to install this exact version via pip typing 

> pip install git+https://github.com/deepmipt/fastText.git#egg=fastText==0.8.22

-After that i tried rerunning `build_model(configs.ner.ner_rus)` and got an error with pandas: `AttributeError: module 'pandas.core.computation' has no attribute 'expressions'`. I was able to solve it by upgrading dask: `pip install dask --upgrade`

Hope it will be useful.

**Comment by yoptar** [comment]
This should have been resolved after merging #855.
Please open a new issue if this problem is still relevant.
